src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -271,3 +352,265 @@ def on_load_checkpoint(self, checkpoint: dict[str, Any]) -> None:
+        # https://github.com/openvinotoolkit/anomalib/issues/1752
+        transform = transform or self.transform or self.configure_transforms()
+        inference_model = InferenceModel(model=self.model, transform=transform, disable_antialias=True)
+        export_root = _create_export_root(export_root, export_type)
+        self._write_metadata_to_json(export_root, task)
+        onnx_path = export_root / "model.onnx"
+        torch.onnx.export(
+            inference_model,
+            torch.zeros((1, 3, 1, 1)).to(self.device),
+            str(onnx_path),
+            opset_version=14,
+            dynamic_axes={"input": {0: "batch_size", 2: "height", 3: "weight"}, "output": {0: "batch_size"}},
+            input_names=["input"],
+            output_names=["output"],
+        )
+
+        return onnx_path
+
+    def to_openvino(
+        self,
+        export_root: Path | str,
+        transform: Transform | None = None,
+        ov_args: dict[str, Any] | None = None,
+        task: TaskType | None = None,
+    ) -> Path:
+        """Convert onnx model to OpenVINO IR.
+
+        Args:
+            export_root (Path): Path to the export folder.
+            transform (Transform, optional): Input transforms used for the model. If not provided, the transform is
+                taken from the model.
+                Defaults to ``None``.
+            ov_args: Model optimizer arguments for OpenVINO model conversion.
+                Defaults to ``None``.
+            task (TaskType | None): Task type.
+                Defaults to ``None``.
+
+        Returns:
+            Path: Path to the exported onnx model.
+
+        Raises:
+            ModuleNotFoundError: If OpenVINO is not installed.
+
+        Returns:
+            Path: Path to the exported OpenVINO IR.
+
+        Examples:
+            Export the Lightning Model to OpenVINO IR:
+            This example demonstrates how to export the Lightning Model to OpenVINO IR.
+
+            >>> from anomalib.models import Patchcore
+            >>> from anomalib.data import Visa
+            ...
+            >>> datamodule = Visa()
+            >>> model = Patchcore()
+            ...
+            >>> model.to_openvino(
+            ...     export_root="path/to/export",
+            ...     transform=datamodule.test_data.transform,
+            ...     task=datamodule.test_data.task
+            ... )
+
+            Using Custom Transforms:
+            This example shows how to use a custom ``Transform`` object for the ``transform`` argument.
+
+            >>> from torchvision.transforms.v2 import Resize
+            >>> transform = Resize(224, 224)
+            ...
+            >>> model.to_openvino(
+            ...     export_root="path/to/export",
+            ...     transform=transform,
+            ...     task="segmentation",
+            ... )
+
+        """
+        model_path = self.to_onnx(export_root, transform, task, ExportType.OPENVINO)
```

Comment:
Don't we actually want to enforce the `ExportType` to be `ONNX` when exporting to onnx? It seems very counterintuitive to call `to_onnx` with `OpenVINO` as export type. I guess this was done to prevent creating an `onnx` subdirectory when calling openvino export, but I'm not sure if this is the right approach, as it would lead to inconsistent placement of the onnx weights:

`model.to_onnx()` -> onnx weights are placed in `onnx` subdirectory
`model.to_openvino()` -> onnx weights are placed in `openvino` subdirectory

I would expect that regardless of export type, onnx weights are always placed in `onnx` subdirectory, and `openvino` subdirectory only contains the OpenVINO weights.

Reply:
this could be temporary though. OpenVINO now supports direct conversion without ONNX, so we may not need it anymore.

Reply:
Exporting model to OpenVINO is now straightforward. I had to add the `example_input` argument when converting the model since the conversion from TorchScript to OpenVINO via `scripting` failed. Fortunately, conversion via `tracing` worked.
https://github.com/openvinotoolkit/anomalib/blob/0244625ba7ea2eaedd444d765889165a63f45903/src/anomalib/deploy/export.py#L282-L290
https://github.com/openvinotoolkit/openvino/blob/0799163b6189213775bea5a38ab03dfe55431fbd/src/bindings/python/src/openvino/frontend/pytorch/ts_decoder.py#L50-L56